src/shogun/evaluation/GradientResult.h

```diff
@@ -92,20 +92,20 @@ class CGradientResult : public CEvaluationResult
 
 		for (index_t i=0; i<m_gradient->get_num_elements(); i++)
 		{
-			CMapNode<TParameter*, SGVector<float64_t> >* param_node=
+			CMapNode<AnyParameter*, SGVector<float64_t> >* param_node=
 				m_gradient->get_node_ptr(i);
 
 			// get parameter name
-			const char* param_name=param_node->key->m_name;
+			const char* param_name=param_node->key->get_properties().get_name().c_str();
 
 			// get object name
-			const char* object_name=
-				m_parameter_dictionary->get_element(param_node->key)->get_name();
+//			const char* object_name=
+//				m_parameter_dictionary->get_element(param_node->key)->get_name();
```

Comment:
Here I need the name of the object the parameter belongs to. AnyParameter doesn't have this information so I will have to think of a way to make it work.